data/tutorials/wf_01_debugging.md

```diff
@@ -392,3 +393,139 @@ Raised at Stdlib__List.assoc in file "list.ml", line 191, characters 10-25
 Called from Dune__exe__Uncaught.find_address in file "uncaught.ml" (inlined), line 3, characters 24-42
 Called from Dune__exe__Uncaught in file "uncaught.ml", line 8, characters 15-37
 ```
+
+
+## Detecting a Data Race with Thread Sanitizer
+
+With the introduction of multicore parallelism in OCaml 5, comes the
+risk of introducing data races among the involved `Domain`s. Luckily
+the Thread Sanitizer (TSan) mode for OCaml is helpful to catch and
+report these.
+
+### Installing a TSan Switch
+
+To install the TSan mode
+1. Install the [`libunwind`](https://github.com/libunwind/libunwind)
+   dependency. On macOS `libunwind` should already be installed by default.
+   On a Linux system with the `apt` package manager, installing
+   `libunwind` should be as simple as `sudo apt install libunwind-dev`.
+2. Create a TSan switch by running `opam switch create 5.0.0+tsan`.
+
+To confirm that the TSan switch is installed correctly, run `opam
+switch show` and confirm that it prints `5.0.0+tsan`.
+
+### Detecting a Data Race
+
+Now consider the following OCaml program written in the file `race.ml`:
+
+```ocaml
+(* file race.ml *)
+type t = { mutable x : int }
+
+let v = { x = 0 }
+
+let () =
+  let t1 = Domain.spawn (fun () -> v.x <- 10; Unix.sleep 1) in
+  let t2 = Domain.spawn (fun () -> v.x <- 11; Unix.sleep 1) in
+  Domain.join t1;
+  Domain.join t2;
+  Printf.printf "v.x is %i\n" v.x
+```
+
+It builds a record `v` with a mutable field `x` initialized to `0`.
+Next, it spawns two parallel `Domain`s `t1` and `t2` that both update
+the field `v.x`.
+
+Here is a corresponding `dune` file:
+```dune
+(executable
+ (name race)
+ (modules race)
+ (libraries unix))
+```
+
+If we compile and run the program using `dune` under a regular `5.0.0` switch the
+program appears to work:
+```
+$ dune build ./race.exe
+$ dune exec ./race.exe
+v.x is 11
+```
+
+However if we compile and run the program with `dune` from the new
```

Comment:
```suggestion
However, if we compile and run the program with Dune from the new
```
If it's the build system in general rather than a `dune` file, I suggest capitalising it